### PR TITLE
Compare github id in a case-insensitive way; closes #3794

### LINKED ISSUE #3794: Can't start work on reserved issue.
**Describe the bug**
Can't start work on reserved issue; no button to start work is shown.

**To Reproduce**
Have someone make a new issue reserved for a user.

**Expected behavior**
Should see a "Start work" or "Express interest" button.

**Screenshots**
https://i.imgur.com/dwjM60A.png

- OS: any
- Browser: all

### PR DESCRIPTION
**Pull request by wanseob**
Github handles are usually used in a case-insensitive way

##### Description

It closes issue #3794. Users who use capital letters in their Github IDs, had a problem that sometimes the bounty detail page does not show "Express interest" button because it recognizes 'StevenJNPearce' and 'stevenjnpearce' as different accounts.

##### Checklist

- [x] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [x] Linter status: 100% pass
- [ ] Added relevant tests
- [x] Tests pass and code coverage has not decreased
- [x] Changes don't break existing behavior
- [x] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [ ] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [ ] Completed: Code review by core team and any requested changes

##### Refers/Fixes

Fixes: #3794 

##### Testing and Sign-off

**Before**
screenshot from 2019-03-04 00-03-05

**After**
screenshot from 2019-03-04 00-03-23

###### Contributor

- [x] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Tested all changes **locally**
- [x] Verified existing functionality
- [x] Ran `make test` and everything passed!

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging or local deployment

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

File: app/assets/v2/js/pages/bounty_details.js [hunk 1]
```diff
@@ -65,6 +65,15 @@ const token_value_to_display = function(val) {
   return Math.round((parseInt(val) / Math.pow(10, document.decimals)) * 1000) / 1000;
 };
 
+const case_insensitive_cmp = function(val1, val2) {
+  if (val1 && val2) {
+    if (typeof val1 === 'string' && typeof val2 === 'string') {
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_details.js]: Can we mage this into 1 if statement + make it into an arrow functions ?

Comment (wanseob) [reply app/assets/v2/js/pages/bounty_details.js]: Thanks for your review @thelostone-mc. I add a new commit for it. Should I squash them?

File: app/assets/v2/js/pages/bounty_details.js [hunk 2]
```diff
@@ -65,6 +65,13 @@ const token_value_to_display = function(val) {
   return Math.round((parseInt(val) / Math.pow(10, document.decimals)) * 1000) / 1000;
 };
 
+const case_insensitive_cmp = (val1, val2) => {
```
Comment (owocki) [inline app/assets/v2/js/pages/bounty_details.js]: this could probably live in shared.js...

Comment (wanseob) [reply app/assets/v2/js/pages/bounty_details.js]: I agree with that idea!

**Comment by wanseob** [comment]
@owocki, @thelostone-mc `caseInsensitiveCompare` method has been moved to "shared.js". And checked that works as expected. Appreciate your feedbacks.

**Comment by thelostone-mc** [comment]
Thanks @wanseob for fixing this up :)

**Pull request merged**